test/ex_cmd/stream_test.exs

```diff
@@ -0,0 +1,30 @@
+defmodule ExCmd.StreamTest do
+  use ExUnit.Case, async: true
+
+  test "simple stream input" do
+    stream =
+      1..10
+      |> Stream.map(&to_string/1)
+      |> Stream.take(5)
+
+    assert ["12345"] =
+             ExCmd.stream!(~w(cat), input: stream)
+             |> Enum.to_list()
```

Comment:
I think there is race condition here. The reader and writer are independent. The reader might read partial output, for example read `123` first, and `45` on next read. 

Locally, test randomly fails like this:
```
  1) test simple stream input (ExCmd.StreamTest)
     test/ex_cmd/stream_test.exs:4
     match (=) failed
     code:  assert ["12345"] = ExCmd.stream!(~w"cat", input: stream) |> Enum.to_list()
     left:  ["12345"]
     right: ["1", "2", "3", "4", "5"]
     stacktrace:
       test/ex_cmd/stream_test.exs:10: (test)
```

It is better to assert the complete output rather than as a list.

```suggestion
    assert "12345" =
             ExCmd.stream!(~w(cat), input: stream)
             |> Enum.into("")
```

Reply:
Ahh yes, great catch.  Swapped to above and ran the tests repeatedly and seems ok.